cpp/src/arrow/compute/kernels/vector_selection_internal.cc

```diff
@@ -863,6 +920,22 @@ Status DenseUnionFilterExec(KernelContext* ctx, const ExecSpan& batch, ExecResul
   return FilterExec<DenseUnionSelectionImpl>(ctx, batch, out);
 }
 
+Status SparseUnionFilterExec(KernelContext* ctx, const ExecSpan& batch, ExecResult* out) {
```

Comment:
Nit: move this into `vector_filter_internal.cc` along `StructFilterExec`? (can probably also share some code between them...)

Reply:
Moved and extracted a `FilterWithTakeExec`.